chain/db/stateGenerator.go

```diff
@@ -120,6 +121,25 @@ func (cs *ChainStore) generateStateRoot(b *block.Block, genesisBlockInitialized,
 
 	//process previous block
 	height := b.Header.UnsignedHeader.Height
+	if height == 0 {
+		preBlockHash := EmptyUint256
+		txnHash := EmptyUint256
+		data := append(preBlockHash[:], txnHash[:]...)
+		data = append(data, b.Header.UnsignedHeader.RandomBeacon...)
+		id := crypto.Sha256(data)
+
+		pk, err := crypto.NewPubKeyFromBytes(b.Header.UnsignedHeader.Signer)
+		if err != nil {
+			return nil, EmptyUint256, err
+		}
+		contract, err := contract.CreateSignatureContract(pk)
+		if err != nil {
+			return nil, EmptyUint256, err
+		}
+
+		states.UpdateID(contract.ProgramHash, id)
```

Comment:
We should either handle error here (and below), or remove the error returned by UpdateID (because it always returns nil now)